### DISCUSSION TITLE
Require MetaData

### DISCUSSION THREAD

**Discussion by BimleshTrigyn** [Sunbird-Ed/Community#469] [Bugs] [answered]
Hi , user is asking the meta-data of  attached excel-sheet do_id’s .

Medium 

Grade  

Subject  

DigitalTextbook  

Chapter  

I've checked on the web portal , and through postman-api-call also , but few are the meta-data are not available, please see below.

Medium → is available.

Grade → (GradeLevel) is available

Subject → subject is available

DigitalTextbook → not available.

Chapter → not available.

can you please check the un-available meta-data, and if it’s not there . then can you please implement it.

 

Thanks
[List of DO IDs (1).xlsx](https://github.com/Sunbird-Ed/Community/files/11273239/List.of.DO.IDs.1.xlsx)

**Comment by kameshbhr** [comment]
@BimleshTrigyn  Please provide the following details:
1. What is the version of Sunbird Ed you are using?
2. What is the API that you are trying to use and what is the input and what is the output?

You can also refer to [this ]( https://knowlg.sunbird.org/learn/product-and-developer-guide/assets-search-service/faqs#what-are-the-different-modes-that-can-be-passed-in-search-request ) page to see the examples of using search API.
"Sample 14 - Find published Textbooks to which a content is linked to" in that page seems to be the example of what you are looking for.

Please raise such queries as part of "General Discussions", not as Bugs.

**Reply by BimleshTrigyn** [reply]
Hii @kameshbhr ,

Thanks for the response.


I'm using content read Api , to fetch the meta-data of particular content-id, Do-id.
Api : https://staging.sunbirded.org/api/content/v2/read/:Content_Id

As , you've suggested to use composite search-Api , can you you please guide , how I can get the meta-data of any particular content-id, 
Do-id... or list of content-id, Do-id. ?

requesting you to please provide an example by taking any content-id, Do-id from attached excel-sheet.

Thanks.

**Reply by kameshbhr** [reply, marked as answer]
@BimleshTrigyn Information about to which collection (like a Digital Textbook) a content is linked to is not part of Content Metadata. So you will not get that information from the Content Read API.  A content can be linked to one or more collections. So you need to use Search API to get that details.

The page I had shared provides the example filters to be used for the search query to get the Digital Textbook details:
{ "request": { "filters": { "primaryCategory":"Digital Textbook", "leafNodes": {"contains": <<do_Id_content>>} }, "sort_by":{"name":"asc"} } }

So to get the information that you want, you need to use both the queries - Read Content API and Search API.

@vinukumar-vs @amitpriyadarshi @pallakartheekreddy
